src/shaders/raster.fragment.glsl

```diff
@@ -12,18 +12,46 @@ uniform float u_saturation_factor;
 uniform float u_contrast_factor;
 uniform vec3 u_spin_weights;
 
+#ifdef RASTER_COLOR
+uniform sampler2D u_color_ramp;
+uniform vec2 u_colorization_scale;
+uniform vec4 u_colorization_mix;
+
+vec4 colormap (float value) {
+  float scaled_value = value * u_colorization_scale.y + u_colorization_scale.x;
+  return texture2D(u_color_ramp, vec2(scaled_value, 0.5));
+}
+#endif
+
 void main() {
 
     // read and cross-fade colors from the main and parent tiles
     vec4 color0 = texture2D(u_image0, v_pos0);
     vec4 color1 = texture2D(u_image1, v_pos1);
+
+#ifdef RASTER_COLOR
+    // In the case of raster colorization, interpolate the raster value first,
+    // then sample the color map. Otherwise we interpolate two tabulated colors
+    // and end up with a color *not* on the color map.
+    vec4 color = colormap(dot(vec4(mix(color0.rgb, color1.rgb, u_fade_t), 1), u_colorization_mix));
```

Comment:
Please move the definition of `vec4 color` before the `#ifdef` because we convert these blocks to regular if block in Metal. (Eventually we'll add some checks for the gl-js repo for these formatting requirements.)

So:
```
vec4 color;
#ifdef RASTER_COLOR
    color = ...
```
And the same variable can be used in the else branch.